internal/resource/config/localidentity/local_identity_identity_profiles_resource.go

```diff
@@ -0,0 +1,917 @@
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.RegistrationEnabled) {
+		addRequest.RegistrationEnabled = plan.RegistrationEnabled.ValueBoolPointer()
+	}
+
+	if internaltypes.IsDefined(plan.RegistrationConfig) {
+		addRequest.RegistrationConfig = client.NewRegistrationConfigWithDefaults()
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.RegistrationConfig, true)), addRequest.RegistrationConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.ProfileConfig) {
+		addRequest.ProfileConfig = client.NewProfileConfigWithDefaults()
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.ProfileConfig, false)), addRequest.ProfileConfig)
+		fmt.Println(err)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.FieldConfig) {
+		addRequest.FieldConfig = client.NewFieldConfig()
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.FieldConfig, false)), addRequest.FieldConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.EmailVerificationConfig) {
+		addRequest.EmailVerificationConfig = client.NewEmailVerificationConfigWithDefaults()
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.EmailVerificationConfig, true)), addRequest.EmailVerificationConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsNonEmptyObj(plan.DataStoreConfig) {
+		addRequest.DataStoreConfig = client.NewLdapDataStoreConfigWithDefaults()
+		err := json.Unmarshal([]byte(internaljson.FromValue(plan.DataStoreConfig, false)), addRequest.DataStoreConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	if internaltypes.IsDefined(plan.ProfileEnabled) {
+		addRequest.ProfileEnabled = plan.ProfileEnabled.ValueBoolPointer()
+	}
+	return nil
+}
+
+// Metadata returns the resource type name.
+func (r *localIdentityIdentityProfilesResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_local_identity_identity_profiles"
+}
+
+func (r *localIdentityIdentityProfilesResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+func (r *localIdentityIdentityProfilesResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+
+	var model localIdentityIdentityProfilesResourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+	// Validates Email Verification type for Email Configuration
+	if internaltypes.IsDefined(model.EmailVerificationConfig) {
+		emailVerificationConfig := model.EmailVerificationConfig.Attributes()
```

Comment:
Rather than casting everything it might be easier and more readable if you convert this into the client struct type, so you can access the fields directly.

The same applies below when checking other objects' fields